infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -563,8 +564,15 @@ private DetectorTask detectFailure(PollReport pollReport) {
             return DETECTOR_TASK_SKIPPED;
         }
 
+        ClusterState clusterState = clusterContext.getClusterView();
```

Comment:
You would want to shift this check just after the first if condition:
```if (localMonitoringService.getMetrics().join().getSequencerStatus() == SequencerStatus.READY)```

This is because we do not want to increment the sequencerNotReadyCounter if the primary sequencer is already bootstrapped.

Reply:
Since now you are already checking if the primary sequencer is bootstrapped for a particular epoch, you can also remove the first if condition check ```if (localMonitoringService.getMetrics().join().getSequencerStatus() == SequencerStatus.READY)```

Reply:
Thank you so much for the insights.